src/engraving/rendering/dev/beamlayout.cpp

```diff
@@ -1247,7 +1258,27 @@ void BeamLayout::createBeamSegment(Beam* item, ChordRest* startCr, ChordRest* en
             }
 
             int extraBeamAdjust = cr->up() ? beamsBelow : beamsAbove;
-            addition = grow * (level - extraBeamAdjust) * item->beamDist();
+            int stemShortenFactor = 0;
```

Comment:
I think moving this to a separate, dedicated function would make things more readable (this function is already too long as it is). Actually, I think also the whole stem extension part of this function should be moved out, but we don't have to do that right now. For now you could add a static private function so this becomes something like `int stemShortenFactor = computeStemShortenFactor( ...arguments... )`. The other advantage of making it a separate function is that you can use early returns for the simpler cases and avoid so much nesting.